### PR TITLE
Test/dec 20 kajal

### PR DESCRIPTION
**Pull request by kkaajjaall**

### CODE REVIEW

**Comment by kkaajjaall** [comment]
insert data , update data and view data are done and delete is remain

**Comment by kkaajjaall** [comment]
delete is done  and Add new data to show session message is remain

File: app/Http/Controllers/CourseMajorController.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+namespace App\Http\Controllers;
+
+use App\Models\CourseMajor;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Session;
+
+class CourseMajorController extends Controller
+{
+    public function index(){
+     $CourseMajor = CourseMajor::get();
```
Comment (ithqindia) [inline app/Http/Controllers/CourseMajorController.php:12]: Follow naming convention. This should be $courseMajor

File: app/Http/Controllers/CourseMajorController.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    ]);
+    $course = CourseMajor::get();
+    return view('show',compact('course'));
+}
+
+public function show(Request $request)
+{
+    $course_major =CourseMajor::where('id', $request->id)->get()->first();
+    $user = [];
```
Comment (ithqindia) [inline app/Http/Controllers/CourseMajorController.php:35]: Remove this line

File: app/Http/Controllers/CourseMajorController.php [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        $course_major->is_enabled=(bool)$request->get('is_enabled');
+
+        $course_major->save();
+        return view('success');
+    }
+
+    public function view(Request $request)
+    {
+        $course_major = CourseMajor::find($request->id);
+        return view('Course-major-info', compact('course_major'));
```
Comment (ithqindia) [inline app/Http/Controllers/CourseMajorController.php:59]: follow naming convention course-major-info

File: app/Http/Controllers/CourseMajorController.php [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        $course_major->details= $request->get('details');
+        $course_major->is_enabled=(bool)$request->get('is_enabled');
+
+        $course_major->save();
+        return view('success');
+    }
+
+    public function view(Request $request)
+    {
+        $course_major = CourseMajor::find($request->id);
```
Comment (ithqindia) [inline app/Http/Controllers/CourseMajorController.php:58]: Follow naming convention

File: app/Models/CourseMajor.php [hunk 5]
```diff
@@ -9,7 +9,7 @@
 class CourseMajor extends Model
 {
     use HasFactory;
-
+    protected $table='course_majors';
```
Comment (ithqindia) [inline app/Models/CourseMajor.php:12]: Remove this.

File: routes/web.php [hunk 6]
```diff
@@ -170,3 +172,25 @@
...
+Route::get('/course-major/{id}', [CourseMajorController::class, 'show']);
+Route::put('/course-major', [CourseMajorController::class, 'update']);
+Route::get('/course-major/{id}/view', [CourseMajorController::class, 'view']);
+Route::delete('/course-major/{id}', [CourseMajorController::class, 'destroy']);
+
+// show
+Route::get('/show', function () {
+    $course = CourseMajor::get()->sortBy('name');
+    return view('show', compact('course'));
+});
```
Comment (ithqindia) [inline routes/web.php:191]: Remove this. This should be handled in Route::get('/course-major', [CourseMajorController::class, 'index']);

**Review by ithqindia — CHANGES_REQUESTED** [review]
Format files

File: resources/views/course-major-form.blade.php [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+                            checked>
+                        <label class="form-check-label" for="flexCheckDefault">
+                            is_enabled
+                        </label>
+                    </div>
+                </div>
+                {{-- submit --}}
+                <div class="col-12 mt-5  mb-5">
+                    <button type="submit" class="btn btn-primary" id="permanent_submit"> Submit</button>
+                </div>
```
Comment (ithqindia) [inline resources/views/course-major-form.blade.php:49]: </form></div> missing

File: resources/views/course-majors.blade.php [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+        </div>
+    @endif
+    <table class="table table-rounded table-striped border gy-7 gs-7">
+        <tr class="fw-bold fs-6 text-gray-800 border-bottom border-gray-200 text-center">
+            <th> #</th>
+            <th> Name </th>
+            <th> Enabled</th>
+            <th> Action</th>
+        </tr>
+        @foreach ($courseMajors as $major_course)
```
Comment (ithqindia) [inline resources/views/course-majors.blade.php:26]: wrong variable name $major_course should be $courseMajor

File: resources/views/includes/sidebar/menu.blade.php [hunk 9]
```diff
@@ -81,6 +81,16 @@
...
+            <div class="menu-item">
+                <a class="menu-link" href="/courses-major/add">
+                    <span class="menu-title">Courses major</span>
+                </a>
+            </div>
+            <div class="menu-item">
+                <a class="menu-link" href="/show">
+                    <span class="menu-title">Course major table</span>
+                </a>
+            </div>
```
Comment (ithqindia) [inline resources/views/includes/sidebar/menu.blade.php:93]: Correct the link to show course majors

**Review by ithqindia — CHANGES_REQUESTED** [review]

File: resources/views/includes/sidebar/menu.blade.php [hunk 10]
```diff
@@ -82,7 +82,7 @@
                 </a>
             </div>
             <div class="menu-item">
-                <a class="menu-link" href="/courses-major/add">
+                <a class="menu-link" href="/course-majors/add">
```
Comment (ithqindia) [inline resources/views/includes/sidebar/menu.blade.php:85]: this should be /courses

**Review by ithqindia — CHANGES_REQUESTED** [review]

**Review by ithqindia — APPROVED** [review]

**Comment by ithqindia** [comment]
Merged locally

**Pull request closed without merging**